packages/nuxt/src/app/components/nuxt-link.ts

```diff
@@ -249,16 +249,39 @@ export function defineNuxtLink (options: NuxtLinkOptions) {
           if (!slots.default) {
             return null
           }
-          return slots.default({
+
+          const props = {
             href,
             navigate,
-            route: router.resolve(href!),
             rel,
             target,
             isExternal: isExternal.value,
             isActive: false,
             isExactActive: false
+          }
+
+          Object.defineProperty(props, 'route', {
+            get () {
+              if (!href) { return undefined }
+
+              const url = parseURL(href)
+              return {
+                path: url.pathname,
+                fullPath: url.pathname,
+                query: parseQuery(url.search),
```

Comment:
Even if `.route` accessor is lazy, sub propers (especially this one could be). It might not have significant performance overhead but can also have when there are dozens of components rendering concurrently so probably micro-optimization worth :)

Reply:
The conditions required for this to be hit are that we are 1) using pages module, 2) using a custom link, 3) using an external link, 4) accessing `route` in this slot. I don't see there being dozens of components matching those conditions rendering concurrently; I think in most cases an external link will simply handle the case of the `href` rather than the route object at all.

My thought is that adding the code to handle the slight perf opt of accessing route.query lazily will add more kb to the bundle than is really worth it. But you think this is a much more likely scenario?

Reply:
I agree. But this special kind of component can be used by a CMS / Project and repeated.  Code diff is alow low btw `get query: () => parseQuery`. This kind of optimization is not odd (`error.stack` for instance is also lazy even tough it is not always accessed by users)... So feel free to judge either make change or ignore my comment :p